paasta_tools/secret_providers/vault.py

```diff
@@ -155,6 +155,22 @@ def get_secret_signature_from_data(self, data: Mapping[str, Any]) -> Optional[st
         else:
             return None
 
+    def get_data_from_vault_path(self, path: str) -> Optional[dict]:
+        # clients.read returns None if not set
+        # if it is set, it returns an object with { **metadata, data: {} }
```

Comment:
just double-checking for the benefit of anyone re-using this function in the future: would vault let us store a single string at a given path? e.g., should this be `Optional[Union[Dict, str]]` or something more permissive?

Reply:
hmm let me see -

according to the Vault API docs, this should always return a dict of key value pairs (ie str:str): https://developer.hashicorp.com/vault/api-docs/secret/kv/kv-v1

though there's a chance this just returns an empty dict

can't see a scenario where it would return just [str], so I imagine this is just `Dict`

Reply:
gotcha, in that case `Optional[Dict[str, str]]` might be better than `Optional[dict]` :)